src/hooks/useQueryParam.ts

```diff
@@ -0,0 +1,14 @@
+import { useState, useEffect } from 'react';
+
+const useQueryParams = () => {
+    const [params, setParams] = useState('');
+
+    useEffect(() => {
+        const currentParams = window.location.search;
+        setParams(currentParams);
+    }, []);
+
+    return params;
+};
```

Comment:
This seems useless, instead we should emit Parsed queryParams to make sense out of it